### PR TITLE
Updated validation to allow submission of 'not applicable' in patholo…

### PR DESCRIPTION
**Pull request by hknahal**
…gical_m_category field when ajcc 7th or 8th editions are used.
Related to https://github.com/icgc-argo/argo-dictionary/issues/366

### CODE REVIEW

**Comment by b-f-chan** [comment]
Need to test in QA first

**Comment by hknahal** [comment]
Tested in QA (Dictionary version 121.0)
**Use cases tested**
- [ ] If `pathological_tumour_staging_system` is `AJCC 7th edition` or `AJCC 8th edition`, then `pathological_m_category` is allowed to be `Not applicable`.
ajcc_7_8_eds



- [ ] If `pathological_tumour_staging_system` is `AJCC 6th edition`, then `pathological_m_category` is not allowed to be `Not applicable`. It has to be any of the other accepted values in the controlled terminology.
ajcc_6_ed

**Comment by puneet-oicr** [comment]
@b-f-chan Needs someone assigned for testing.

**Comment by puneet-oicr** [comment]
@b-f-chan Needs someone assigned for testing.

**Comment by b-f-chan** [comment]
@puneet-oicr - @Buwujiu has agreed to take up testing for this

File: tests/common/mxCategoryValidation.test.js [hunk 1]
```diff
@@ -31,6 +31,28 @@ const primary_diagnosis = require('../constructDummyData').getSchemaDummy('prima
...
+      name: 'pathological_m_category',
+    },
+  ],
+  [
+    'Pathological staging system set to "AJCC 7th edition" and "not applicable" submitted in pathological_m_category',
+    true,
+    {
+      row: loadObjects(specimen, {
+        pathological_tumour_staging_system: 'ajcc 7th edition',
+        pathological_m_category: 'not Applicable'
```
Comment (Buwujiu) [inline tests/common/mxCategoryValidation.test.js:51]: should it be ` 'Not applicable'`?

**Comment by Buwujiu** [comment]
> Tested in DEV and QA (Dictionary version 122.*)
> 
> `pathological_m_category` is only allowed to be `Not applicable` if the `pathological_tumour_staging_system` is `AJCC 7th edition` or `AJCC 8th edition`.
> 
> **Use cases tested** Test case 1:
> 
> * [x]  If `pathological_tumour_staging_system` is `AJCC 7th edition` or `AJCC 8th edition`, and `pathological_m_category` is `Not applicable`, the validation should pass.
>   Test files:
>   [donor.txt](https://github.com/icgc-argo/argo-dictionary/files/10117998/donor.txt)
>   [primary_diagnosis.txt](https://github.com/icgc-argo/argo-dictionary/files/10117999/primary_diagnosis.txt)
>   [specimen.txt](https://github.com/icgc-argo/argo-dictionary/files/10118000/specimen.txt)
> 
> ajcc_7_8_eds
> 
> Test case 2:
> 
> * [x]  If `pathological_tumour_staging_system` is `AJCC 6th edition`, and `pathological_m_category` is `Not applicable`, the validation should fail. `pathological_m_category`must be any of the other accepted values in the controlled terminology.
> 
> Test case 3:
> 
> * [x]  If `pathological_tumour_staging_system` is `AJCC 7th edition` or `AJCC 8th edition` and `pathological_m_category` is `MX`, the validation should fail. `pathological_m_category` is not allowed to be `MX` if AJCC 7th or 8th editions are used.
> 
> Test files for Test case 2 & 3 [donor.txt](https://github.com/icgc-argo/argo-dictionary/files/10118018/donor.txt) [primary_diagnosis.txt](https://github.com/icgc-argo/argo-dictionary/files/10118019/primary_diagnosis.txt) [specimen.txt](https://github.com/icgc-argo/argo-dictionary/files/10118020/specimen.txt) 370_mx_not_allowed

Tested the above cases in ALEXIS-INTL in qa, working as expected!

File: tests/common/mxCategoryValidation.test.js (hunk 1 above)
Comment (hknahal) [reply tests/common/mxCategoryValidation.test.js:51]: For testing, we want to validate regardless of upper/lower case. So even if `Not applicable` is submitted as a mix of upper/lower case letters, as long as the spelling is correct, it should validate.

**Review by Buwujiu — APPROVED** [review]

**Pull request merged**